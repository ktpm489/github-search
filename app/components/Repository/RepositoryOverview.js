import React from 'react';
import PropTypes from 'prop-types';
import { View, Text, StyleSheet, Platform } from 'react-native';
import { Button } from 'react-native-elements';
import { graphql } from 'react-apollo';

import trackRepositoryName from '../../graphql/trackRepositoryName';

export class RepositoryOverview extends React.Component {
  static propTypes = {
    navigation: PropTypes.object.isRequired,
    handlePress: PropTypes.func.isRequired,
    name: PropTypes.string.isRequired,
    description: PropTypes.string,
  };

  handleTitlePress = () => {
    const { navigation, handlePress, name } = this.props;
    handlePress(name);
    navigation.navigate('RepositoryDetailScreen', { repositoryName: name });
  };

  render() {
    const { name, description } = this.props;

    return (
      <View style={styles.container}>
        <Button
          title={name}
          small
          onPress={this.handleTitlePress}
          textStyle={[styles.repoName, styles.fontColor, styles.flexWrap]}
          color="#222"
          backgroundColor="#FFF"
          buttonStyle={styles.button}
          containerViewStyle={styles.buttonContainer}
        />
        <View style={styles.repoDescriptionContainer}>
          <Text style={[styles.fontColor, styles.flexWrap]}>{description}</Text>
        </View>
      </View>
    );
  }
}

export default graphql(trackRepositoryName, {
  props: ({ mutate }) => ({
    handlePress: (name) => mutate({ variables: { name } }),
  }),
})(RepositoryOverview);

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'flex-start',
  },
  repoName: {
    fontFamily: Platform.OS === 'ios' ? 'HelveticaNeue' : 'Roboto',
    fontWeight: Platform.OS === 'ios' ? '500' : '700',
    fontSize: 20,
  },
  repoDescriptionContainer: {
    paddingTop: 3,
    paddingBottom: 5,
    alignSelf: 'stretch',
  },
  fontColor: {
    color: '#222',
  },
  flexWrap: {
    flexWrap: 'wrap',
  },
  button: {
    paddingLeft: 0,
    paddingBottom: 3,
  },
  buttonContainer: {
    marginLeft: 0,
    marginRight: 0,
    maxWidth: '100%',
  },
});
