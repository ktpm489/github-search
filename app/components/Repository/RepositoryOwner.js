import React from 'react';
import PropTypes from 'prop-types';
import { View, Text, StyleSheet } from 'react-native';
import { Avatar } from 'react-native-elements';

const RepositoryOwner = ({
  avatarUrl,
  name,
  login,
  height,
  width,
  containerStyle,
  flexDirection,
  infoContainerStyle,
  onAvatarPress,
}) => (
  <View style={[styles.container, { flexDirection }]}>
    <Avatar
      height={height}
      width={width}
      rounded
      source={{ uri: avatarUrl }}
      containerStyle={containerStyle}
      onPress={onAvatarPress}
    />
    <View style={infoContainerStyle}>
      <Text style={styles.name}>{name}</Text>
      <Text style={styles.login}>{login}</Text>
    </View>
  </View>
);

RepositoryOwner.propTypes = {
  avatarUrl: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  login: PropTypes.string.isRequired,
  height: PropTypes.number,
  width: PropTypes.number,
  containerStyle: PropTypes.oneOfType([PropTypes.number, PropTypes.object]),
  flexDirection: PropTypes.oneOf(['row', 'column']),
  infoContainerStyle: PropTypes.oneOfType([PropTypes.number, PropTypes.object]),
  onAvatarPress: PropTypes.func.isRequired,
};

RepositoryOwner.defaultProps = {
  containerStyle: {},
  height: 100,
  width: 100,
  flexDirection: 'column',
  infoContainerStyle: {},
};

const styles = StyleSheet.create({
  container: {
    display: 'flex',
    alignItems: 'center',
    marginBottom: 10,
  },
  infoContainer: {
    display: 'flex',
    flex: 1,
    marginLeft: 10,
    justifyContent: 'center',
  },
  name: {
    color: '#000',
    fontSize: 22,
    fontWeight: '700',
  },
  login: {
    color: '#6C7680',
    fontSize: 18,
  },
});

export default RepositoryOwner;
