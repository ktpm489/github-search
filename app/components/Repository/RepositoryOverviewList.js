import React from 'react';
import PropTypes from 'prop-types';
import { FlatList, StyleSheet, View } from 'react-native';

// eslint-disable-next-line import/no-named-as-default
import RepositoryOverview from './RepositoryOverview';

class RepositoryOverviewList extends React.Component {
  static propTypes = {
    repos: PropTypes.arrayOf(
      PropTypes.shape({
        node: PropTypes.shape({
          name: PropTypes.string.isRequired,
          description: PropTypes.string,
          url: PropTypes.string.isRequired,
        }),
      })
    ).isRequired,
    navigation: PropTypes.object.isRequired,
    onEndReached: PropTypes.func.isRequired,
  };

  renderItem = ({
    item: {
      node: { name, description, url },
    },
  }) => {
    const { navigation } = this.props;
    return (
      <RepositoryOverview
        name={name}
        description={description}
        url={url}
        navigation={navigation}
      />
    );
  };

  render() {
    const { repos, onEndReached } = this.props;
    return (
      <FlatList
        style={{ flex: 1 }}
        data={repos}
        renderItem={(repo) => this.renderItem(repo)}
        keyExtractor={(item, index) => index}
        ItemSeparatorComponent={() => <View style={styles.separator} />}
        onEndReached={onEndReached}
      />
    );
  }
}

const styles = StyleSheet.create({
  separator: {
    height: 1,
    backgroundColor: '#222',
    flex: 1,
    flexDirection: 'row',
  },
});

export default RepositoryOverviewList;
