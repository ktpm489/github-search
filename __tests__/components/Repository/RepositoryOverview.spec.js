import React from 'react';
import { shallow } from 'enzyme';
import { Text } from 'react-native';
import { Button } from 'react-native-elements';

import { RepositoryOverview } from '../../../app/components/Repository/RepositoryOverview';

describe('<RepositoryOverview />', () => {
  it('renders correctly with all required props', () => {
    const wrapper = shallow(
      <RepositoryOverview
        name="github-search"
        description="A React Native app for searching Github users."
        navigation={{ navigate: jest.fn() }}
        handlePress={jest.fn()}
      />
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(Text).props().children).toEqual(
      'A React Native app for searching Github users.'
    );
    expect(wrapper.find(Button).props().title).toEqual('github-search');
  });

  it('executes handlePress when the button is pressed', () => {
    const navigate = jest.fn();
    const handlePress = jest.fn();

    const wrapper = shallow(
      <RepositoryOverview
        name="github-search"
        description="A React Native app for searching Github users."
        navigation={{ navigate }}
        handlePress={handlePress}
      />
    );

    const component = wrapper.instance();
    component.handleTitlePress();
    expect(handlePress).toHaveBeenCalledWith('github-search');
    expect(navigate).toHaveBeenCalledWith('RepositoryDetailScreen', {
      repositoryName: 'github-search',
    });
  });
});
