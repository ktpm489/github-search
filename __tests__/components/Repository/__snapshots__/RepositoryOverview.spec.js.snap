// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RepositoryOverview /> renders correctly with all required props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RepositoryOverview
    description="A React Native app for searching Github users."
    handlePress={[MockFunction]}
    name="github-search"
    navigation={
      Object {
        "navigate": [MockFunction],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Button
          backgroundColor="#FFF"
          buttonStyle={
            Object {
              "paddingBottom": 3,
              "paddingLeft": 0,
            }
          }
          color="#222"
          containerViewStyle={
            Object {
              "marginLeft": 0,
              "marginRight": 0,
              "maxWidth": "100%",
            }
          }
          iconRight={
            Object {
              "color": "#222",
              "name": "call-made",
            }
          }
          onPress={[Function]}
          small={true}
          textStyle={
            Array [
              Object {
                "fontFamily": "HelveticaNeue",
                "fontSize": 20,
                "fontWeight": "500",
              },
              Object {
                "color": "#222",
              },
              Object {
                "flexWrap": "wrap",
              },
            ]
          }
          title="github-search"
        />,
        <View
          style={
            Object {
              "alignSelf": "stretch",
              "borderTopColor": "#222",
              "borderTopWidth": 1,
              "paddingTop": 3,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#222",
                },
                Object {
                  "flexWrap": "wrap",
                },
              ]
            }
          >
            A React Native app for searching Github users.
          </Text>
        </View>,
      ],
      "style": Object {
        "alignItems": "flex-start",
        "flex": 1,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "backgroundColor": "#FFF",
          "buttonStyle": Object {
            "paddingBottom": 3,
            "paddingLeft": 0,
          },
          "color": "#222",
          "containerViewStyle": Object {
            "marginLeft": 0,
            "marginRight": 0,
            "maxWidth": "100%",
          },
          "iconRight": Object {
            "color": "#222",
            "name": "call-made",
          },
          "onPress": [Function],
          "small": true,
          "textStyle": Array [
            Object {
              "fontFamily": "HelveticaNeue",
              "fontSize": 20,
              "fontWeight": "500",
            },
            Object {
              "color": "#222",
            },
            Object {
              "flexWrap": "wrap",
            },
          ],
          "title": "github-search",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#222",
                },
                Object {
                  "flexWrap": "wrap",
                },
              ]
            }
          >
            A React Native app for searching Github users.
          </Text>,
          "style": Object {
            "alignSelf": "stretch",
            "borderTopColor": "#222",
            "borderTopWidth": 1,
            "paddingTop": 3,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "A React Native app for searching Github users.",
            "ellipsizeMode": "tail",
            "style": Array [
              Object {
                "color": "#222",
              },
              Object {
                "flexWrap": "wrap",
              },
            ],
          },
          "ref": null,
          "rendered": "A React Native app for searching Github users.",
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Button
            backgroundColor="#FFF"
            buttonStyle={
              Object {
                "paddingBottom": 3,
                "paddingLeft": 0,
              }
            }
            color="#222"
            containerViewStyle={
              Object {
                "marginLeft": 0,
                "marginRight": 0,
                "maxWidth": "100%",
              }
            }
            iconRight={
              Object {
                "color": "#222",
                "name": "call-made",
              }
            }
            onPress={[Function]}
            small={true}
            textStyle={
              Array [
                Object {
                  "fontFamily": "HelveticaNeue",
                  "fontSize": 20,
                  "fontWeight": "500",
                },
                Object {
                  "color": "#222",
                },
                Object {
                  "flexWrap": "wrap",
                },
              ]
            }
            title="github-search"
          />,
          <View
            style={
              Object {
                "alignSelf": "stretch",
                "borderTopColor": "#222",
                "borderTopWidth": 1,
                "paddingTop": 3,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#222",
                  },
                  Object {
                    "flexWrap": "wrap",
                  },
                ]
              }
            >
              A React Native app for searching Github users.
            </Text>
          </View>,
        ],
        "style": Object {
          "alignItems": "flex-start",
          "flex": 1,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "backgroundColor": "#FFF",
            "buttonStyle": Object {
              "paddingBottom": 3,
              "paddingLeft": 0,
            },
            "color": "#222",
            "containerViewStyle": Object {
              "marginLeft": 0,
              "marginRight": 0,
              "maxWidth": "100%",
            },
            "iconRight": Object {
              "color": "#222",
              "name": "call-made",
            },
            "onPress": [Function],
            "small": true,
            "textStyle": Array [
              Object {
                "fontFamily": "HelveticaNeue",
                "fontSize": 20,
                "fontWeight": "500",
              },
              Object {
                "color": "#222",
              },
              Object {
                "flexWrap": "wrap",
              },
            ],
            "title": "github-search",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#222",
                  },
                  Object {
                    "flexWrap": "wrap",
                  },
                ]
              }
            >
              A React Native app for searching Github users.
            </Text>,
            "style": Object {
              "alignSelf": "stretch",
              "borderTopColor": "#222",
              "borderTopWidth": 1,
              "paddingTop": 3,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "A React Native app for searching Github users.",
              "ellipsizeMode": "tail",
              "style": Array [
                Object {
                  "color": "#222",
                },
                Object {
                  "flexWrap": "wrap",
                },
              ],
            },
            "ref": null,
            "rendered": "A React Native app for searching Github users.",
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
